plugins/module_utils/mso.py

```diff
@@ -280,6 +280,24 @@ def ndo_bfd_multi_hop_settings_spec():
     )
 
 
+def ndo_template_object_spec():
+    return dict(
+        type="dict",
+        options=dict(
+            name=dict(type="str"),
+            template=dict(type="str"),
+            template_id=dict(type="str"),
+        ),
+        required_by={
+            "template": "name",
+            "template_id": "name",
+        },
+        mutually_exclusive=[
+            ["template", "template_id"],
+        ],
+    )
```

Comment:
Shouldn't we allow the option to add aliases to the name of the object in the spec:
```suggestion
def ndo_template_object_spec(aliases=None):
    return dict(
        type="dict",
        options=dict(
            name=dict(type="str", aliases=aliases) if aliases else dict(type="str"),
            template=dict(type="str"),
            template_id=dict(type="str"),
        ),
        required_by={
            "template": "name",
            "template_id": "name",
        },
        mutually_exclusive=[
            ["template", "template_id"],
        ],
    )
```

Reply:
Done